packages/muelu/src/Rebalancing/MueLu_RepartitionFactory_def.hpp

```diff
@@ -101,12 +101,14 @@ namespace MueLu {
     Input(currentLevel, "Partition");
   }
 
-  template<class T> class MpiTypeTraits            { public: static MPI_Datatype getType(); };
-  template<>        class MpiTypeTraits<long>      { public: static MPI_Datatype getType() { return MPI_LONG;      } };
-  template<>        class MpiTypeTraits<int>       { public: static MPI_Datatype getType() { return MPI_INT;       } };
-  template<>        class MpiTypeTraits<short>     { public: static MPI_Datatype getType() { return MPI_SHORT;     } };
-  template<>        class MpiTypeTraits<unsigned>  { public: static MPI_Datatype getType() { return MPI_UNSIGNED;  } };
-  template<>        class MpiTypeTraits<long long> { public: static MPI_Datatype getType() { return MPI_LONG_LONG; } };
+  template<class T> class MpiTypeTraits                     { public: static MPI_Datatype getType(); };
+  template<>        class MpiTypeTraits<long>               { public: static MPI_Datatype getType() { return MPI_LONG;      } };
+  template<>        class MpiTypeTraits<int>                { public: static MPI_Datatype getType() { return MPI_INT;       } };
+  template<>        class MpiTypeTraits<short>              { public: static MPI_Datatype getType() { return MPI_SHORT;     } };
+  template<>        class MpiTypeTraits<unsigned>           { public: static MPI_Datatype getType() { return MPI_UNSIGNED;  } };
+  template<>        class MpiTypeTraits<long long>          { public: static MPI_Datatype getType() { return MPI_LONG_LONG; } };
+  template<>        class MpiTypeTraits<unsigned long>      { public: static MPI_Datatype getType() { return MPI_UNSIGNED_LONG; } };
+  template<>        class MpiTypeTraits<unsigned long long> { public: static MPI_Datatype getType() { return MPI_UNSIGNED_LONG_LONG; } };
```

Comment:
Support for `long long` in MPI might depend on `MPI_VERSION`; be sure to check the MPI standard.

Reply:
I checked the standards. It is missing in 1.3, is optional in 2.0, and fully present since 2.1. Do you think that's an issue? I guess I could follow Teuchos and add `if MPI_VERSION >= 2`.

Reply:
@aprokop Thanks for checking! :-) I'm pretty sure Trilinos explicitly requires `MPI_VERSION >= 2` (I mean the MPI standard version, not the OpenMPI software implementation version -- I know you know, but I want to avoid confusion for readers).  You could always just add a compile-time check for `MPI_VERSION >= 2`, just to help the odd user who doesn't realize they are pointing to an old MPI version.